### PR TITLE
Add process environment setting support

### PR DESCRIPTION
**Pull request by pangqingyuan1991**
Hi,
I have recreate a new one.

### CODE REVIEW

**Comment by pangqingyuan1991** [comment]
I notice CFLAGS has -DQT_NO_CAST_FROM_ASCII, but I cannot find which file define this flag. This is the reason I can't use QString("abcd") directly

File: src/perfrecord.cpp [hunk 1]
```diff
@@ -172,6 +172,13 @@ void PerfRecord::startRecording(const QStringList& perfOptions, const QString& o
     }
     m_perfRecordProcess = new QProcess(this);
     m_perfRecordProcess->setProcessChannelMode(QProcess::MergedChannels);
+    auto e = QProcessEnvironment::systemEnvironment();
+    foreach(const QString &str, env){
```
Comment (milianw) [inline src/perfrecord.cpp]: for (const auto &str : env) {

File: src/perfrecord.cpp [hunk 2]
```diff
@@ -172,6 +172,13 @@ void PerfRecord::startRecording(const QStringList& perfOptions, const QString& o
...
+        QStringList var = str.split(QChar::fromLatin1('='));
```
Comment (milianw) [inline src/perfrecord.cpp]: const auto var = str.split(QLatin1Char('='));

Comment (milianw) [reply src/perfrecord.cpp]: hm no, that's actually not enough, when the (string) value contains another = then we don't want to split again, so this should find the first `=` via indexOf and then pass the first and last part to `e.insert`

File: src/perfrecord.cpp [hunk 3]
```diff
@@ -172,6 +172,13 @@ void PerfRecord::startRecording(const QStringList& perfOptions, const QString& o
     }
     m_perfRecordProcess = new QProcess(this);
     m_perfRecordProcess->setProcessChannelMode(QProcess::MergedChannels);
+    auto e = QProcessEnvironment::systemEnvironment();
```
Comment (milianw) [inline src/perfrecord.cpp]: please give it a more descriptive name, like `processEnvironment`

File: src/perfrecord.cpp [hunk 4]
```diff
@@ -172,6 +172,13 @@ void PerfRecord::startRecording(const QStringList& perfOptions, const QString& o
...
+    foreach(const QString &str, env){
+        QStringList var = str.split(QChar::fromLatin1('='));
+        if(var.length()!=2) continue;
```
Comment (milianw) [inline src/perfrecord.cpp]: style:
```
if (var.size() != 2)
    continue;
```

File: src/recordpage.ui [hunk 5]
```diff
@@ -117,13 +117,39 @@
...
+      <item row="2" column="1">
+       <widget class="QLineEdit" name="applicationEnvBox">
+        <property name="contextMenuPolicy">
+         <enum>Qt::NoContextMenu</enum>
+        </property>
+        <property name="toolTip">
+         <string>Optional environments to be set for the application being recorded</string>
+        </property>
+        <property name="placeholderText">
+         <string>FOO=1 BOO=2</string>
```
Comment (milianw) [inline src/recordpage.ui]: please use `FOO=1 BAR="asdf"` :)

File: src/recordpage.ui [hunk 6]
```diff
@@ -117,13 +117,39 @@
...
+        </property>
+       </widget>
+      </item>
+      <item row="2" column="1">
+       <widget class="QLineEdit" name="applicationEnvBox">
+        <property name="contextMenuPolicy">
+         <enum>Qt::NoContextMenu</enum>
+        </property>
+        <property name="toolTip">
+         <string>Optional environments to be set for the application being recorded</string>
```
Comment (milianw) [inline src/recordpage.ui]: Optional environment *variables*

**Comment by pangqingyuan1991** [comment]
> pretty close, thanks for re-instantiating this patch! It's valuable and I'd like to get it in. can you fix the nits? otherwise I'll do that for you.
> 
> And yes, I'm setting `-DQT_NO_CAST_FROM_ASCII` deliberately to prevent the performance pitfalls of accidentally allocating utf16 QString data from utf8 string literals.

Hi, I think this is just a little feature and no much to do. So you can change it yourself.

**Comment by milianw** [comment]
ok, I'll see when I get around to do that.

cheers

**Pull request closed without merging**